generatebundlefile/ecr_helper.go

```diff
@@ -186,7 +186,16 @@ func getLatestHelmTagandSha(details []ImageDetailsBothECR) (string, string, erro
 	if reflect.DeepEqual(latest, ImageDetailsBothECR{}) {
 		return "", "", fmt.Errorf("error no images found")
 	}
-	return latest.ImageTags[0], *latest.ImageDigest, nil
+
+	var imageTag string
+	imageTagRegex := regexp.MustCompile(`^\d+\.\d+\.\d+-[0-9a-f]{40}$`)
+	for _, tag := range latest.ImageTags {
+		if imageTagRegex.MatchString(tag) {
```

Comment:
nit: if possible it will be nice to also filter out a latest-helm tag from a commit cos i saw ran into this yesterday where the private ECR has images with tags both `0.7.1-eks-1-27-32-fe14b98f98132123f478eaf6e65bba4a29bf488b` and `0.7.1-eks-1-27-32-fe14b98f98132123f478eaf6e65bba4a29bf488b-latest-helm` this and during the promote it somehow ends up promoting the one with suffix latest-helm.

Reply:
The regex should filter out the latest helm because it will only accept tags which end at the SHA (note the `$` at the end of the regex)